### PR TITLE
Ports/mrsh: Fix workdir, remove or upgrade patches

### PR DESCRIPTION
**Pull request by EWouters**
The patches to `fnmatch` and `ulimit` no longer seem necessary (please correct me if I'm wrong). The new patches are in the preferred style and have a ReadMe.md, so mrsh was also removed from the list of ports missing descriptions in the linter. 

Seems to work, even though I don't know mrsh: 
image

### CODE REVIEW

File: Ports/mrsh/package.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env -S bash ../.port_include.sh
 port=mrsh
 version=cd3c3a48055ab4085d83f149ff4b4feba40b40cb
-files="https://codeload.github.com/emersion/mrsh/legacy.tar.gz/${version} emersion-mrsh-d9763a3.tar.gz 6896493a1020774715ccca28e8d8f4ec722af63a93543fb6dd2762f7b1de9c8a"
+files="https://api.github.com/repos/emersion/mrsh/tarball/${version} ${port}-${version}.tar.gz 03fa30993c8cc9eab4b10092f278ace847de60484ac2ea21537c2b1922f09f9e"
```
Comment (timschumi) [inline Ports/mrsh/package.sh:4]: Can't we just use the well-known `archive` links, i.e. `https://github.com/emersion/mrsh/archive/${version}.tar.gz`?

File: Ports/mrsh/patches/0003-Workaround-warnings.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+index 0f433d2..577afde 100644
+--- a/main.c
++++ b/main.c
+@@ -14,7 +14,9 @@
+ #include <unistd.h>
+ #include "frontend.h"
+ 
++#ifndef __serenity__
+ extern char **environ;
++#endif
```
Comment (timschumi) [inline Ports/mrsh/patches/0003-Workaround-warnings.patch:20]: Is this actually needed? Why would it complain about an `extern` definition of `environ`?

File: Ports/mrsh/patches/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## `0003-Workaround-warnings.patch`
+
+Workaround warnings
+
+
+## `0004-Disable-glob.patch`
+
+Disable glob
+
+
```
Comment (timschumi) [inline Ports/mrsh/patches/ReadMe.md:22]: The patches could probably use a bit more of a description. Why are we disabling glob? What warnings do we workaround? What's wrong with the default version script?

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/mrsh/patches/0003-Workaround-warnings.patch (hunk 2 above)
Comment (EWouters) [reply Ports/mrsh/patches/0003-Workaround-warnings.patch:20]: This is the error is remedies:
```
main.c:17:15: error: redundant redeclaration of 'environ' [-Werror=redundant-decls]
   17 | extern char **environ;
      |               ^~~~~~~
In file included from main.c:14:
/Users/erik/git/serenity-master/Build/x86_64/Root/usr/include/unistd.h:31:15: note: previous declaration of 'environ' with type 'char **'
   31 | extern char** environ;
      |               ^~~~~~~
cc1: all warnings being treated as errors
```
I'll be more expressive in the commit message.

File: Ports/mrsh/patches/ReadMe.md (hunk 3 above)
Comment (EWouters) [reply Ports/mrsh/patches/ReadMe.md]: I tried to be a bit more descriptive about the patches, but as I picked this port up from
someone else, I'm not 100% sure how to best explain the patches.

File: Ports/mrsh/package.sh [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
-workdir=emersion-mrsh-d9763a3
 
-export CFLAGS=-Wno-deprecated-declarations
+pre_configure() {
+    export CFLAGS=-Wno-deprecated-declarations
+}
+
+post_configure() {
+    unset CFLAGS
+}
```
Comment (timschumi) [inline Ports/mrsh/package.sh:14]: Is this actually needed? Why can't we just keep the general `export CFLAGS=...` as before?

Comment (timschumi) [inline Ports/mrsh/package.sh:14]: Missing newline at the end of the file.

**Review by timschumi — CHANGES_REQUESTED** [review]
Two more small things.

File: Ports/mrsh/package.sh (hunk 4 above)
Comment (EWouters) [reply Ports/mrsh/package.sh]: This is not needed, but I've seen other packages do it this way, and I thought it was cleaner. If you say it is unneeded I'll revert to how it was.

Comment (EWouters) [reply Ports/mrsh/package.sh]: I've deleted the extra code. It wasn't doing anything usefull.

**Review by timschumi — APPROVED** [review]
Looks reasonable, and runs fine on `{i686,x86_64} GNU`.

**Pull request merged**